src/functions/edi/inbound/handler.ts

```diff
@@ -102,34 +102,50 @@ export const handler = async (event: any): Promise<Record<string, any>> => {
             receivingPartnerId,
           });
 
-          for (const functionalGroup of interchange.functionalGroups) {
-            const guideIdsForPartnership = transactionSetConfigs.map(
-              (config) => config.guideId
-            );
+          const guideIdsForPartnership = transactionSetConfigs.map(
+            (config) => config.guideId
+          );
 
+          for (const functionalGroup of interchange.functionalGroups) {
             // For each Transaction Set:
-            // - look up the guideId and mappingId
-            // - call processTransactionSet, which translates X12 to JSON, and then invokes the mapping
-            // - send the result to the webhook
-            // - delete the input file once processed successfully
-
+            // - look up the guideId
+            // - find the corresponding transaction set config from the partnership
+            // - call processEdi, which translates X12 to JSON
+            // - call deliverToDestination for each destination, which does the following:
+            //   - optionally invokes the mapping if one is included in config
+            //   - sends the result to the destination
             for (const transactionSet of functionalGroup.transactionSets) {
               // load the guide for the transaction set
               const guideSummary = await resolveGuide({
                 guideIdsForPartnership,
                 transactionSetType: transactionSet.transactionSetType,
               });
 
-              const edi = transactionSet.segments.join(
+              const transactionSetContents = transactionSet.segments.join(
                 interchange.delimiters.segment
               );
 
+              const documentContents = [
+                interchange.segments.ISA,
+                functionalGroup.segments.GS,
+                transactionSetContents,
+                functionalGroup.segments.GE,
+                interchange.segments.IEA,
+              ];
+
+              const edi = documentContents.join(interchange.delimiters.segment);
+
               // find the transaction set config for partnership that includes guide
               const transactionSetConfig = resolveTransactionSetConfig(
                 transactionSetConfigs,
                 guideSummary.guideId
               );
 
+              const ediJson = await processEdi(
```

Comment:
we used to translate/map for every destination. now there's only 1 translate, with N mapping operations (optionally one per destination)